docs/kibana.asciidoc

```diff
@@ -0,0 +1,232 @@
+
+It is quite straightforward to connect a Kibana instance to an Elasticsearch cluster managed by ECK:
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: kibana.k8s.elastic.co/{eck_crd_version}
+kind: Kibana
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRef:
+    name: quickstart
+    namespace: default
+----
+
+NOTE: `namespace` is optional if the Elasticsearch cluster is running in the same namespace as Kibana.
+
+The Kibana configuration file is automatically setup by ECK to establish a secure connection to Elasticsearch.
+
+[float]
+[id="{p}-kibana-advanced-configuration"]
+=== Advanced configuration
+
+If you already have read the documentation about how to run Elasticsearch on ECK then most of the concepts and the ideas
+described here will sound familiar to you.
+This is because the resource definitions in ECK share the same philosophy when it comes to:
+
+* Customize the Pod configuration
+* Customize the product configuration
+* Manage HTTP settings
+* Use secure settings
+
+[float]
+[id="{p}-kibana-pod-configuration"]
+==== Pod Configuration
+You can define a <<{p}-pod-template,pod template>> to customize the Kibana pod and override any configuration values.
+
+The following example demonstrates how to create a Kibana deployment with custom node affinity and resource limits.
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: kibana.k8s.elastic.co/{eck_crd_version}
+kind: Kibana
+metadata:
+  name: kibana-sample
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRef:
+    name: "elasticsearch-sample"
+  podTemplate:
+    spec:
+      containers:
+      - name: kibana
+        resources:
+          requests:
+            memory: 1Gi
+            cpu: 0.5
+          limits:
+            memory: 2Gi
+            cpu: 2
+      nodeSelector:
+        type: frontend
+----
+
+NOTE: The name of the container in the pod template must be `kibana`.
+
+See <<{p}-compute-resources-kibana-and-apm>> for more information.
+
+[float]
+[id="{p}-kibana-configuration"]
+==== Kibana Configuration
+You can add your own Kibana settings to the `spec.config` section.
+
+The following example demonstrates how to set the `elasticsearch.requestHeadersWhitelist` configuration option:
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: kibana.k8s.elastic.co/{eck_crd_version}
+kind: Kibana
+metadata:
+  name: kibana-sample
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRef:
+    name: "elasticsearch-sample"
+  config:
+     elasticsearch.requestHeadersWhitelist:
+     - authorization
+----
+
+NOTE: The configuration items you provide always take precedence over the ones that are generated by the operator.
+
+[float]
+[id="{p}-kibana-scaling"]
+=== Scale out a Kibana deployment
+
+You may want to deploy more than one instance of Kibana. In this case all the instances have to share
+the same encryption key.
+
+This can be done either directly in the `spec.config` section:
```

Comment:
I wonder if we should make it clearer that this is not the recommended way to do it

Reply:
> LGTM, a couple of nits. Another thing I realized we might want to call out is how to connect to an external cluster, e.g.
> [#1875 (comment)](https://github.com/elastic/cloud-on-k8s/issues/1875#issuecomment-539886642)
> 
> because it's easy to configure it insecurely (for instance, putting the ES password in the Kibana spec rather than a secret), so it would be helpful to give people a clear and secure example

👍 , and this is what I wanted to do at first when I realized that user must provide a reference to an Elasticsearch cluster. I'm not sure it makes sense to talk about that until #1875 is resolved.

Reply:
👍 I have remove the example, not sure it is a good idea to explain how to do insecure things.